managed/services/telemetry/telemetry.go

```diff
@@ -106,18 +106,35 @@ func (s *Service) Run(ctx context.Context) {
 	defer ticker.Stop()
 
 	doSend := func() {
+		var settings *models.Settings
+		err := s.db.InTransaction(func(tx *reform.TX) error {
+			var e error
+			if settings, e = models.GetSettings(tx); e != nil {
+				return e
+			}
+			return nil
+		})
+		if settings.Telemetry.Disabled {
+			s.l.Info("disabled via settings")
+			return
+		}
+
 		report, err := s.prepareReport(ctx)
 		if err != nil {
 			s.l.Debugf("Failed to prepare report: %s.", err)
 			return
 		}
 
-		err = s.send(ctx, report)
-		if err != nil {
-			s.l.Debugf("Telemetry info not sent, due to error: %s.", err)
-			return
+		if s.config.Reporting.Send {
```

Comment:
what is the difference between `s.config` and `settings` ?
Why some options are checked in on source, the rest in the other ?

Reply:
settings - from DB
s.config - service config